### PR TITLE
Ignore FunctionOnlyReturningConstant for allowed annotations

### PR DESCRIPTION
**Pull request by sowmyav24**
Fixes [#1313](https://github.com/arturbosch/detekt/issues/1313)

- Introduced `excludeAnnotatedFunctions` to provide a list of annotations that can disable `FunctionOnlyReturningConstant` check
- Set default value to `dagger.Provides`

### CODE REVIEW

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/FunctionOnlyReturningConstantSpec.kt [hunk 1]
```diff
@@ -31,6 +31,22 @@ class FunctionOnlyReturningConstantSpec : Spek({
             val rule = FunctionOnlyReturningConstant(config)
             assertThat(rule.compileAndLint(code)).hasSize(0)
         }
+
+        it("does not report excluded annotated function which returns a constant") {
+            val code = """
+                import jdk.nashorn.internal.ir.annotations.Ignore
```
Comment (schalkms) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/FunctionOnlyReturningConstantSpec.kt]: This is the reason why the build fails on some machines when compiling the test snippet. Please remove this import here.

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/FunctionOnlyReturningConstantSpec.kt [hunk 2]
```diff
@@ -31,6 +31,22 @@ class FunctionOnlyReturningConstantSpec : Spek({
...
+                class Test {
+                    @Ignore
+                    fun someIgnoredFun(): String {
+                        return "I am a constant"
+                    }
+                }
+            """.trimIndent()
+            val config = TestConfig(mapOf("excludeAnnotatedFunction" to "jdk.nashorn.internal.ir.annotations.Ignore"))
+            val rule = FunctionOnlyReturningConstant(config)
+            assertThat(rule.compileAndLint(code)).hasSize(0)
```
Comment (schalkms) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/FunctionOnlyReturningConstantSpec.kt]: nit

```suggestion
            assertThat(rule.compileAndLint(code)).isEmpty()
```

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/FunctionOnlyReturningConstantSpec.kt (hunk 1 above)
Comment (schalkms) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/FunctionOnlyReturningConstantSpec.kt]: You could use the `kotlin.SinceKotlin` annotation for testing purposes.

**Comment by schalkms** [comment]
Thanks! I like this config option.

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/FunctionOnlyReturningConstantSpec.kt (hunk 1 above)
Comment (sowmyav24) [reply detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/FunctionOnlyReturningConstantSpec.kt]: Sure, Done.

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/FunctionOnlyReturningConstantSpec.kt (hunk 2 above)
Comment (sowmyav24) [reply detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/FunctionOnlyReturningConstantSpec.kt]: Done !

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/FunctionOnlyReturningConstantSpec.kt (hunk 1 above)
Comment (sowmyav24) [reply detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/FunctionOnlyReturningConstantSpec.kt]: Yes, replaced with `kotlin.SinceKotlin`

**Pull request merged**